app/population/middleware/set-population-id.js

```diff
@@ -0,0 +1,24 @@
+const locationsFreeSpacesService = require('../../../common/services/locations-free-spaces')
+
+async function setResultsPopulationTable(req, res, next) {
```

Comment:
With [router.param middleware](https://expressjs.com/en/api.html#router.param) you can also use the 4th argument as the value. So this could become:

```javascript
async function setPopulationId(req, res, next, locationId) {
  ...
}
```

Reply:
This middleware needs both the passed in param and the date param which is not passed in - so there's not a lot of point using one as an argument, and then retrieving the other direct from req.params